fluent-react/src/provider.ts

```diff
@@ -27,10 +29,21 @@ interface LocalizationProviderProps {
 export function LocalizationProvider(
   props: LocalizationProviderProps
 ): ReactElement {
+  let [locales, setLocales] = useState(props.initialLocales);
+
+  function changeLocales(locales: string[]) {
+    props.changeLocales(locales);
+    setLocales(locales);
```

Comment:
There aren't any check if the `prop.changeLocales` was succeeded before to call `setLocales`. Maybe we could return a bool on `changeLocales`? Or use `await props.changeLocales(locales)` in order to fail on except?